WordPress/Classes/Utility/Editor/EditorFactory.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+/// This class takes care of instantiating the correct editor based on the App settings, feature flags,
+/// etc.
+///
+class EditorFactory {
+
+    /// Settings for the Gutenberg logic.
+    ///
+    private let gutenbergSettings = GutenbergSettings()
+
+    // MARK: - Editor: Instantiation
+
+    /// We can't return a type that's both a PostEditor and a UIViewController, so using
+    /// a configure block as a hack.
+    /// In Swift 4, we'll be able to do `instantiateEditor() -> UIViewController & PostEditor`,
+    /// and then let the caller configure the editor.
+    func instantiatePostEditor(post: AbstractPost, configure: (PostEditor, UIViewController) -> Void) -> UIViewController {
```

Comment:
We are using Swift 4 in the main App from some time already, maybe we can take this refactor as an opportunity to update this section?